lib/remote/apilistener-configsync.cpp

```diff
@@ -113,6 +144,14 @@ Value ApiListener::ConfigUpdateObjectAPIHandler(const MessageOrigin::Ptr& origin
 	if (!object && !config.IsEmpty()) {
 		newObject = true;
 
+		Defer unlockAndNotifyOne([&lock, &ptype, &objName]{
+			m_ObjectNames[ptype.get()].erase(objName);
+			// Release the mutex before notifying a thread, otherwise it will be blocked immediately after being awakened.
+			lock.unlock();
+
+			m_ObjectNamesCV.notify_one();
+		});
+
```

Comment:
Why all this? I'd just unconditionally _Defer_, like when deleting below.

Reply:
But I want to notify as soon as the new object is successfully created or failing and not waiting for the entire method to complete.

Reply:
What's the rest of the method? Updating an already existing object? Not sure if doing that concurrently for the same object is a good idea. At least only performing only one config update for an object at the same time sounds like a sensible idea.

Reply:
> What's the rest of the method? Updating an already existing object?

Only if the received object version is newer than the existing one.